### PR TITLE
[ACR] Support Overriding Task parameters, support for ContinuationToken [NEW preview API VERSION]

### PR DESCRIPTION
**Pull request by shahzzzam**
Introduces the API version `2019-06-01-preview` for ACR Tasks.

- `OverrideTaskStepProperties` for a `TaskRunRequest` - You can now override some properties when you created the Task. But you can now also override them.
- `updateTriggerToken, updateTriggerEndpoint, updateTriggerPayloadType` properties for Break-the-loop
- `taskName` -> `taskId` as a required property for Task run.
- Adds `arm64` and `386` to list of supported platforms.

### Diff against previous swagger
You can diff it against `stable/2019-04-01` swagger.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by shahzzzam** [comment]
@tjprescott is there any issue going on with azure-sdk-for-python and azure-sdk-for-ruby right now? Locally, I was able to create the SDK for python without any issues.
// cc: @jaysterp @northtyphoon

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6345 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6345/commits/95b3ba544acb36d09f3f2d6f41d11579bbe7e2fe

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (kpajdzik) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]:
```suggestion
            "description": "Error response describing why the operation failed. If the registry doesn't exist, 404 (Not found) is returned. If any of the input parameters is wrong, 400 (Bad Request) is returned."
```

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (kpajdzik) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]:
```suggestion
            "description": "Error response describing why the operation failed. If the registry doesn't exist, 404 (Not found) is returned. If any of the input parameters is wrong, 400 (Bad Request) is returned."
```

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (kpajdzik) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: Can we do min and max here?

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (kpajdzik) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: Is it possible to share it with enum on line 943?

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (kpajdzik) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]:
```suggestion
          "description": "The URL (absolute or relative) of the source context. It can be an URL to a tar or git repository.\r\nIf it is relative URL, the relative path should be obtained from calling listBuildSourceUploadUrl API.",
```

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (kpajdzik) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: Can you use this definition - https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json#L9 ?

**Comment by jaysterp** [comment]
@shahzzzam can you `includeTriggerMetadata` in your examples so we can see the expected response?

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (RyanBensonMSFT) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: We should avoid booleans in swagger, we typically use string enums as they're extensible and forward compatible.
Consider something more along the lines of an optional property that looks like this:
'updateTriggerContent' : ['triggerMetadata']

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: @jaysterp  seems like such changes will also update all the previous swaggers, has this not been caught before - or was ignored bc of the regressive swagger updates?

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: @jaysterp same as above

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: @northtyphoon do we have a min and max here?

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: good catch! i missed adding it to that one. Both of these needs to be still different properties tho.
`PlatformProperties` is passed while creating a Task.
`PlatformUpdateProperties` is passed while updating a Task.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: @kpajdzik can these changes be ignored? Do I need to update all the previous swaggers for this?
cc: @jaysterp

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: @kpajdzik I dont think we have a min and max here.
cc @northtyphoon

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (kpajdzik) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: It's a minor issue that's going to impact documentation only so it can be ignored. You don't have to fix previous swaggers.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (sanjaiganesh) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: Compared to the previous version, there are lot of apis that got dropped.. Could you add details on why those got dropped to the description. ?
for ex: collection get call on following URi is not present.. & /CheckNameAvailability is dropped.
GET /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}

[https://portal.azure-devex-tools.com/tools/diff/eyJzb3VyY2VVcmwiOiJodHRwczovL2dpdGh1Yi5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvYmxvYi9tYXN0ZXIvc3BlY2lmaWNhdGlvbi9jb250YWluZXJyZWdpc3RyeS9yZXNvdXJjZS1tYW5hZ2VyL01pY3Jvc29mdC5Db250YWluZXJSZWdpc3RyeS9zdGFibGUvMjAxOS0wNS0wMS9jb250YWluZXJyZWdpc3RyeS5qc29uIiwidGFyZ2V0VXJsIjoiaHR0cHM6Ly9naXRodWIuY29tL0F6dXJlL2F6dXJlLXJlc3QtYXBpLXNwZWNzL2Jsb2IvYTlhYjU1NzNhMmY1OTk2MzVlNjMwYzYwOWEzYmU4NzY0MDFhZGFmZi9zcGVjaWZpY2F0aW9uL2NvbnRhaW5lcnJlZ2lzdHJ5L3Jlc291cmNlLW1hbmFnZXIvTWljcm9zb2Z0LkNvbnRhaW5lclJlZ2lzdHJ5L3N0YWJsZS8yMDE5LTA2LTAxL2NvbnRhaW5lcnJlZ2lzdHJ5X2J1aWxkLmpzb24ifQ==](https://portal.azure-devex-tools.com/tools/diff/eyJzb3VyY2VVcmwiOiJodHRwczovL2dpdGh1Yi5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvYmxvYi9tYXN0ZXIvc3BlY2lmaWNhdGlvbi9jb250YWluZXJyZWdpc3RyeS9yZXNvdXJjZS1tYW5hZ2VyL01pY3Jvc29mdC5Db250YWluZXJSZWdpc3RyeS9zdGFibGUvMjAxOS0wNS0wMS9jb250YWluZXJyZWdpc3RyeS5qc29uIiwidGFyZ2V0VXJsIjoiaHR0cHM6Ly9naXRodWIuY29tL0F6dXJlL2F6dXJlLXJlc3QtYXBpLXNwZWNzL2Jsb2IvYTlhYjU1NzNhMmY1OTk2MzVlNjMwYzYwOWEzYmU4NzY0MDFhZGFmZi9zcGVjaWZpY2F0aW9uL2NvbnRhaW5lcnJlZ2lzdHJ5L3Jlc291cmNlLW1hbmFnZXIvTWljcm9zb2Z0LkNvbnRhaW5lclJlZ2lzdHJ5L3N0YWJsZS8yMDE5LTA2LTAxL2NvbnRhaW5lcnJlZ2lzdHJ5X2J1aWxkLmpzb24ifQ==)

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: Hi @sanjaiganesh , the previous API version is `2019-04-01` for ACR Tasks. 
Some of the API versions belong to ACR RP (for instance, 2019-05-01 that you compared), and some of them belong to ACR Task RP (previous version 2019-04-01). 
cc: @jaysterp in case she can better explain this.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: done, converted to an enum 'updateTriggerPayloadType' as seen in the new swagger

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: Not sure, This has been in the previous API versions too. @jaysterp do you know about this?

Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-06-01/containerregistry_build.json]: not sure, @jaysterp do you know why?

**Comment by shahzzzam** [comment]
@jaysterp  when u have time, can u help me figure out why ModelValidations are failing for this..

**Comment by shahzzzam** [comment]
@jaysterp @northtyphoon please review.

**Review by northtyphoon — APPROVED** [review]
LGTM

**Comment by shahzzzam** [comment]
@kpajdzik  can you please review again?

**Review by kpajdzik — APPROVED** [review]
LGTM although there is plenty of breaking changes. Waiting for ARM feedback before merging.

File: specification/containerregistry/resource-manager/readme.md
Comment (ankurkhemani) [inline specification/containerregistry/resource-manager/readme.md]: Why do we need this ?

**Comment by shahzzzam** [comment]
@kpajdzik what are the breaking changes you refer to?

**Review by ankurkhemani — APPROVED** [review]

File: specification/containerregistry/resource-manager/readme.md
Comment (ankurkhemani) [reply specification/containerregistry/resource-manager/readme.md]: I think it's not really important unless you want to consume it as default api version here. Don't have a very strong opinion though, upto you :) 
LGTM.

Comment (shahzzzam) [reply specification/containerregistry/resource-manager/readme.md]: I removed it. Thanks

**Comment by shahzzzam** [comment]
@kpajdzik @RyanBensonMSFT @sanjaiganesh can you please merge this if there are no issues. My team has approved the changes. There are no breaking changes. So if @kpajdzik could elaborate on that. Thanks.

**Comment by kpajdzik** [comment]
> @kpajdzik @RyanBensonMSFT @sanjaiganesh can you please merge this if there are no issues. My team has approved the changes. Thanks.

I was comparing with https://github.com/Azure/azure-rest-api-specs/blob/master/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json this spec. Is that the latest version?

**Comment by shahzzzam** [comment]
@kpajdzik no, the previous version is: 
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-04-01/containerregistry_build.json

Let me know :)

**Comment by kpajdzik** [comment]
> @kpajdzik no, the previous version is:
> https://github.com/Azure/azure-rest-api-specs/blob/master/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-04-01/containerregistry_build.json
> 
> Let me know :)

My bad. The only breaking change is property change in `TaskRunRequest`. 

Have you seen those kind if errors in ModelValidation?
```

 error : ‌
operationId: Tasks_Update
scenario: Tasks_Update_WithMSICustomCredentials
source: request
responseCode: ALL
severity: 0
code: READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST
details:
  code: READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST
  params:
    - properties
    - Docker
  message: 'ReadOnly property `"properties": "Docker"`, cannot be sent in the request.'
  path: properties/step/type
  title: '#/definitions/TaskStepUpdateParameters/properties/type'
  description: The type of the step.
  jsonPath: $.properties.step.type
  position:
    line: 1766
    column: 17
  url: >-
    specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
  directives: {}
```

**Comment by shahzzzam** [comment]
@kpajdzik Thanks. Fixed it. Let me know. The other ModelValidation issue seems to be a problem from rest-api-specs side. See https://github.com/Azure/azure-rest-api-specs/pull/6221

**Comment by kpajdzik** [comment]
> @kpajdzik Thanks. Fixed it. Let me know. The other ModelValidation issue seems to be a problem from rest-api-specs side. See #6221
> 
> If it looks okay, can you merge it? Or @ someone from ARM who can? Thanks in advance

I'll merge it as soon as ARM team approves it. @RyanBensonMSFT, any updates here?

**Comment by shahzzzam** [comment]
@RyanBensonMSFT did you get a chance to review this? We kind of want to get this out in the next AZ CLI release (with new Python SDK). Thanks!

**Comment by shahzzzam** [comment]
@kpajdzik anyone else you know that can approve this? it's been two days.

**Comment by kpajdzik** [comment]
@KrisBash, can this be reassigned to someone else from ARM team?

**Comment by ravbhatnagar** [comment]
@kpajdzik - I will review this from ARM side shortly.

**Comment by ravbhatnagar** [comment]
Looks good. Signing off from ARM side.

**Comment by shahzzzam** [comment]
@kpajdzik Hi Kamil, Can you please merge it?

**Comment by kpajdzik** [comment]
> @kpajdzik Hi Kamil, Can you please merge it?

What about `OBJECT_ADDITIONAL_PROPERTIES` errors in Model Validation?

**Comment by shahzzzam** [comment]
@kpajdzik yup please refer to this: https://github.com/Azure/azure-rest-api-specs/pull/6221

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (amarzavery) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: @shahzzzam - There was a bug in the model validator where it was throwing `INTERNAL_ERROR` instead of the actual error. That bug has been fixed. I just re-run the model validator CI and it is showing correct errors. Please take a look at the CI log over [here]( Tasks_Create_WithSystemAndUserIdentities).

There are basically two types of ERRORs:
1. READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST
2. OBJECT_ADDITIONAL_PROPERTIES

I am still looking into the 2nd type of error to ensure that it is not a false positive.
The 1st error `READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST` is correctly reported and you should be fixing it. 

- The property `"type"` is a discriminator and is not specified in the list of required properties of the model "RunRequest"
A property that is specified as a discriminator **MUST be a required property** of the model. Take a look at the `definition of discriminator` in Open API Specification 2.0 over [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#fixed-fields-13) and `example model definition with discriminator property` over [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#models-with-polymorphism-support).
2.  The discriminator property `"type"` is also marked as `"readOnly": true`. Please read the definition of readOnly over [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#fixed-fields-13). This model is then used in the request body for operation `Registries_ScheduleRun`. The example specifies the discriminator property "type" in the request body and thus the validator fails with `READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST`. If you do not specify the discriminator property then how is the sdk or the user supposed to serialize the model as a Parent `"RunRequest"` or `oneOf` its children `DockerBuildRequest`, `TaskRunRequest`, `FileTaskRunRequest`, `EncodedTaskRunRequest`. This can only be identified by the value provided in the discriminator property `"type"`.

Please make sure that you remove `readOnly: true` for all the properties marked as `discriminator` and add those properties to the list of required properties in the corresponding model definitions.

Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: @amarzavery Thanks for the detailed explanation on this. Quick question tho, the last api version was 2019-04-01. And I just copied over those examples from there. I am not sure how the previous api version did not detect such problems? The changes to the new API version aren't even related to all this.

**Comment by shahzzzam** [comment]
@amarzavery I think the only errors remaining are now `OBJECT_ADDITIONAL_PROPERTIES`. Let me know if something needs to be changed.

**Comment by kpajdzik** [comment]
@amarzavery, Can you admin-merge if everything looks alright to you?

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (amarzavery) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: We improved our validation effort off lately especially the rule to ensure readonly properties are not allowed in the request. That opened the pandora's box of undetected issues.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (amarzavery) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: `"required": [ "type"]` should be present over here since the property is defined in this model and not in the models that have an allOf on this model. Please remove `"type"` from the required property list of all the child model definitions.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/RegistriesScheduleRun_Task.json
Comment (amarzavery) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/RegistriesScheduleRun_Task.json]: Model Validator is throwing the right error. `updateTriggerToken` is a property of  Model `OverrideTaskStepProperties`, whereas here it is defined as a property of runRequest. I think this should be defined after Line 35, inside property `overrideTaskStepProperties`

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (amarzavery) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: This extension is a Mapper for mapping the value on the wire to the model definition defined in the specification. Some services have the value of the discriminator property as `"Microsoft.TaskRunRequest"` whereas the model definition name in the swagger spec is `"TaskRunRequest"`. 

In your case, the name of the model and the value provided in the `"x-ms-discriminator-value"` is the same. Thus please remove this extension from all the places in the spec where it's value and the name of the model are the same.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: hmm, for some reason our swagger generator is putting it in the subclasses, not sure why cc @jaysterp

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/ManagedIdentity/TasksCreate_WithSystemAndUserIdentities.json
Comment (amarzavery) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/ManagedIdentity/TasksCreate_WithSystemAndUserIdentities.json]: The model validator is right again. You did not specify the `"type"` of `"step"`. From the example it seems that the `"type": "Docker"` should be added to `"step"` for the validator to validate against `"DockerBuildStep"` as opposed to `"TaskStepProperties"`.

General comment: Please remove "type" from the required list of properties from the child model definition `"DockerBuildStep"` and add it to `"TaskStepProperties"`. Please make sure you do this for all the Polymorphic models.

Comment (amarzavery) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/ManagedIdentity/TasksCreate_WithSystemAndUserIdentities.json]: @shahzzzam - Please go through each and every error reported by the model validator and fix those issues.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/TasksCreate.json
Comment (amarzavery) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/TasksCreate.json]: missing `"type": "Docker"` for the validator to validate it as `"DockerBuildStep"` instead of `"TaskStepProperties"`. Hence `"arguments"` is an additional property in the current example.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json]: @amarzavery We put `"type"`  as the `[Required]` abstract property on the base class. All the implementation classes don't have a "[Required]" attribute on the `type`. We generate our Swagger using Swashbuckle. And it seems to have added `"required": [ "type" ]` in the swagger of base class + sub classes. 
It seems like this is a limitation of Swashbuckle. What do you suggest we do here? 

Also, note that all tests have passed now.

**Pull request merged**
